app/src/main/kotlin/com/ark/globe/coordinates/URLParser.kt

```diff
@@ -0,0 +1,158 @@
+package com.ark.globe.coordinates
+
+import android.net.Uri
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import java.io.BufferedReader
+import java.net.HttpURLConnection.HTTP_OK
+
+class URLParser {
+    companion object{
+        private const val GOOGLE_MAPS = ".google."
+        private const val GOOGLE_MAPS_SHORT = "goo.gl"
```

Comment:
It's better to add DI and use the repository pattern instead of companion object